src/hotspot/share/gc/g1/g1ConcurrentMark.cpp

```diff
@@ -1106,6 +1106,10 @@ class G1UpdateRemSetTrackingBeforeRebuildTask : public WorkerTask {
 
       bool selected_for_rebuild;
       if (hr->is_humongous()) {
+        oop obj = cast_to_oop(hr->humongous_start_region()->bottom());
+        if (G1CollectedHeap::is_obj_filler(obj)) { // Object allocated, but not well-formed
```

Comment:
Just an idea: I think it would be nicer and more understandable if the code explicitly checked if the object is part of the stalled allocation requests instead of relying on it being a filler object.
It is a bit ugly that we use filler objects to identify objects in construction here, given that "nobody should have a reference to it" (I see the need). It is somewhat okay since it's GC.
Another option would be having dedicated "object-in-construction" objects, which is more work.

Reply:
In the worst case, searching the satisfied allocation requests list might have performance implications. However, that is speculation for now, I haven't tried it.